pkg/remote/v2/oci/resolver.go

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package oci
+
+import (
+	"archive/tar"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	imgname "github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	ociremote "github.com/google/go-containerregistry/pkg/v1/remote"
+	pipelinev1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	"github.com/tektoncd/pipeline/pkg/client/clientset/versioned/scheme"
+	"github.com/tektoncd/pipeline/pkg/remote"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	// KindAnnotation is an OCI annotation for the bundle kind
+	KindAnnotation = "dev.tekton.image.kind"
+	// APIVersionAnnotation is an OCI annotation for the bundle version
+	APIVersionAnnotation = "dev.tekton.image.apiVersion"
+	// TitleAnnotation is an OCI annotation for the bundle title
+	TitleAnnotation = "dev.tekton.image.name"
+	// MaximumBundleObjects defines the maximum number of objects in a bundle
+	MaximumBundleObjects = 20
+)
+
+// Resolver implements the Resolver interface using OCI images.
+type Resolver struct {
```

Comment:
You probably don't need to fork entire impls - it's mainly the interface components that need to be reved.

e.g. the oci.Resolver struct can satisfy both v1.Resolver and v2.Resolver interfaces, so long as those interfaces don't interfere with each other. This will probably be the least invasive way to make this change, particularly since IIUC we only need to change 1 method?

I'd avoid trying to maintain 2 separate copies of implementations to avoid diverging behavior.

Reply:
Sounds good! I refactored it again. Hopefully, its ok this time 🤞